<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>850</width>
    <height>550</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>550</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>900</width>
    <height>550</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Robot Arm</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QSlider" name="horizontalSlider">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>490</y>
      <width>500</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-180</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::NoTicks</enum>
    </property>
   </widget>
   <widget class="QSlider" name="verticalSlider">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>0</y>
      <width>16</width>
      <height>480</height>
     </rect>
    </property>
    <property name="maximum">
     <number>90</number>
    </property>
    <property name="value">
     <number>45</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLabel" name="view_h">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>510</y>
      <width>67</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="view_v">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>480</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>45</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>350</y>
      <width>321</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string>Cartesian</string>
    </property>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>25</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="val_x">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>60</y>
       <width>55</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>30</y>
       <width>25</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="val_y">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>60</y>
       <width>55</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>30</y>
       <width>25</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="val_z">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>60</y>
       <width>55</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>65</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Velocity
(mm/sec)</string>
     </property>
    </widget>
    <widget class="QLabel" name="val_x_v">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>90</y>
       <width>55</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_y_v">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>90</y>
       <width>55</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_z_v">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>90</y>
       <width>55</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>54</y>
       <width>55</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Position
   (mm)</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>0</y>
      <width>321</width>
      <height>351</height>
     </rect>
    </property>
    <property name="title">
     <string>Joint</string>
    </property>
    <widget class="QLineEdit" name="val_J1">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>70</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="slider_J3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>300</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-135</number>
     </property>
     <property name="maximum">
      <number>-1</number>
     </property>
     <property name="value">
      <number>-90</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>1</x>
       <y>120</y>
       <width>62</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Joint 2</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_joint6">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>320</y>
       <width>50</width>
       <height>26</height>
      </rect>
     </property>
     <property name="text">
      <string>Open</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>1</x>
       <y>170</y>
       <width>62</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Joint 3</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="slider_J1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>300</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-90</number>
     </property>
     <property name="maximum">
      <number>90</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="val_J3">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>170</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>-90</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="slider_J2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>300</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>180</number>
     </property>
     <property name="value">
      <number>120</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="val_J5">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>270</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>1</x>
       <y>220</y>
       <width>62</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Joint 4</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="slider_J4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>250</y>
       <width>300</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>90</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>1</x>
       <y>320</y>
       <width>62</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Joint 6</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="slider_J5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>300</y>
       <width>300</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-90</number>
     </property>
     <property name="maximum">
      <number>90</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="val_J4">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>220</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="val_J2">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>120</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>120</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>1</x>
       <y>270</y>
       <width>62</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Joint 5</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>70</y>
       <width>62</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Joint 1</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>27</y>
       <width>60</width>
       <height>38</height>
      </rect>
     </property>
     <property name="text">
      <string>Forward
   (deg)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>27</y>
       <width>51</width>
       <height>38</height>
      </rect>
     </property>
     <property name="text">
      <string>Inverse
  (deg)</string>
     </property>
    </widget>
    <widget class="QLabel" name="val_J1_i">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>70</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_J2_i">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>120</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_J3_i">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>170</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_J4_i">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>220</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_J5_i">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>270</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>27</y>
       <width>71</width>
       <height>38</height>
      </rect>
     </property>
     <property name="text">
      <string>Velocity
(deg/sec)</string>
     </property>
    </widget>
    <widget class="QLabel" name="val_J1_v">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>70</y>
       <width>90</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_J2_v">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>120</y>
       <width>90</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_J3_v">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>170</y>
       <width>90</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_J4_v">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>220</y>
       <width>90</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="val_J5_v">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>270</y>
       <width>90</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>500</width>
      <height>480</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:white</string>
    </property>
    <widget class="QWidget" name="view_widget" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>50</y>
       <width>500</width>
       <height>480</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_gamePad">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>0</y>
       <width>50</width>
       <height>50</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="resource.qrc">
       <normaloff>:/resource/image/resource/image/xbox-dis.png</normaloff>:/resource/image/resource/image/xbox-dis.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>50</width>
       <height>50</height>
      </size>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_home">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>0</y>
       <width>50</width>
       <height>50</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="resource.qrc">
       <normaloff>:/resource/image/resource/image/home.png</normaloff>:/resource/image/resource/image/home.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>50</width>
       <height>50</height>
      </size>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_axis">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>50</width>
       <height>50</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="resource.qrc">
       <normaloff>:/resource/image/resource/image/axis-dis.png</normaloff>:/resource/image/resource/image/axis-dis.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>50</width>
       <height>50</height>
      </size>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_connect">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>0</y>
       <width>50</width>
       <height>50</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="resource.qrc">
       <normaloff>:/resource/image/resource/image/arm-offline.png</normaloff>:/resource/image/resource/image/arm-offline.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>50</width>
       <height>50</height>
      </size>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_record">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>0</y>
       <width>50</width>
       <height>50</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="resource.qrc">
       <normaloff>:/resource/image/resource/image/record.png</normaloff>:/resource/image/resource/image/record.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>50</width>
       <height>50</height>
      </size>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_play">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>0</y>
       <width>50</width>
       <height>50</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="resource.qrc">
       <normaloff>:/resource/image/resource/image/stop.png</normaloff>:/resource/image/resource/image/stop.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>50</width>
       <height>50</height>
      </size>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
    <zorder>btn_gamePad</zorder>
    <zorder>view_widget</zorder>
    <zorder>btn_home</zorder>
    <zorder>btn_axis</zorder>
    <zorder>btn_connect</zorder>
    <zorder>btn_record</zorder>
    <zorder>btn_play</zorder>
   </widget>
   <widget class="QPushButton" name="btn_update">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>480</y>
      <width>80</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Update</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
